core/lib/eth_client/src/clients/mock.rs

```diff
@@ -28,7 +28,16 @@ struct MockTx {
 }
 
 impl From<Vec<u8>> for MockTx {
-    fn from(tx: Vec<u8>) -> Self {
+    fn from(raw_tx: Vec<u8>) -> Self {
+        let is_eip4844 = raw_tx[0] == EIP_4844_TX_TYPE;
+        let tx: Vec<u8> = if is_eip4844 {
+            // decoding rlp-encoded length
+            let len = raw_tx[2] as usize * 256 + raw_tx[3] as usize - 2;
```

Comment:
I've tried to use rlp library for this, but couldn't make it work. This is an rlp-encoded list with 4 elements and I want to extract the first one of it. Do you know how we can decode this instead of hacking it this way?

Reply:
Unfortunately I haven't used rlp lib. I think it's ok to leave it this way for mock client